### PR TITLE
Refacor with_env

### PR DESCRIPTION
**Pull request by Werni**
Move with_env and with_c_locale to helper.rb due to duplicated code.

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged f9776d3760ad40cb603578f7532f3fae0f50c78f into 1039244745d666eb0aa31ebdcf8fd9e5c5cc6936 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged f9776d3760ad40cb603578f7532f3fae0f50c78f into 24e91b9bd6fb0d6223a971775180f16ee3e3b60f )

File: spec/unit/helper_spec.rb [hunk 1]
```diff
@@ -34,3 +34,37 @@
...
+      expect(ENV.include?("MACHINERY_TEST")).to be true
+      expect(ENV.fetch("MACHINERY_TEST")).to eq("bla")
+    end
+  end
+
+  it "resets the environment variables after run" do
+    before_env = ENV
+    with_env "MACHINERY_TEST" => "bla" do
+      expect(ENV.include?("MACHINERY_TEST")).to be true
+    end
```
Comment (thardeck) [inline spec/unit/helper_spec.rb]: expect(ENV.include?("MACHINERY_TEST")).to be false

**Comment by thardeck** [comment]
I am not sure if both methods should be part of the module but it looks a little weird at the moment. Having it in the module would at least make the allocation easier without using "git grep".

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged e676c7a00ddca991965f85b6d8951ec91c74a27b into 24e91b9bd6fb0d6223a971775180f16ee3e3b60f )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 09843bd343a3606ec1b778c67f41c6148d97bf46 into 24e91b9bd6fb0d6223a971775180f16ee3e3b60f )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 09843bd343a3606ec1b778c67f41c6148d97bf46 into b38a4fe7f7c0a8092fcb48d2a66fec9625024a58 )

**Comment by thardeck** [comment]
lgtm

**Pull request merged**